app/src/main/java/com/allan/boardbuddies/fragments/EditBoardFragment.java

```diff
@@ -0,0 +1,116 @@
+package com.allan.boardbuddies.fragments;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewTreeObserver;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.widget.Toolbar;
+import androidx.fragment.app.DialogFragment;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentResultListener;
+import androidx.fragment.app.FragmentTransaction;
+import androidx.lifecycle.ViewModelProvider;
+import androidx.navigation.Navigation;
+
+import com.allan.boardbuddies.Constants;
+import com.allan.boardbuddies.R;
+import com.allan.boardbuddies.fragments.TextDialogFragment;
+import com.allan.boardbuddies.models.Stroke;
+import com.allan.boardbuddies.models.TextBox;
+import com.allan.boardbuddies.viewmodels.EditBoardViewModel;
+import com.allan.boardbuddies.views.CanvasView;
+import com.google.android.material.bottomnavigation.BottomNavigationView;
+import com.google.android.material.navigation.NavigationBarView;
+
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+public class EditBoardFragment extends Fragment implements NavigationBarView.OnItemSelectedListener {
+    private EditBoardViewModel editBoardViewModel;
+    private TextView boardTitle;
+    private CanvasView canvasView;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_edit_board, container, false);
+        Toolbar toolbar = view.findViewById(R.id.board_toolbar);
+
+        boardTitle = view.findViewById(R.id.board_title);
+        canvasView = view.findViewById(R.id.canvas_view);
+        editBoardViewModel = new ViewModelProvider(requireActivity()).get(EditBoardViewModel.class);
+
+        ViewTreeObserver viewTreeObserver = canvasView.getViewTreeObserver();
+        if (viewTreeObserver.isAlive()) {
```

Comment:
curious, when would viewTreeObserver not be alive?

would recommend putting everything after this point into `onViewCreated`, since this is all logic that operates on the created views. you should only be inflating views in `onCreateView`

Reply:
I can probably not have it? It's not alive when the view has been destroyed or just missing...

Reply:
Ok but like what if I never need to access the object again e.g: bind toolbar to view, init toolbar. 
Total 2 usages, but i'd have to make the toolbar a fragment property. Is it worth